From 78ea88e2f3768d5d1d93f1c970a0c6a9315bfb62 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Thu, 15 Nov 2018 12:45:18 +0530
Subject: [PATCH 0488/1852] drm: xlnx: crtc: Fix max width and height overflow

commit 71c63c7a514df68508b8746081f45c80319e4a5e from
https://github.com/Xilinx/linux-xlnx.git

The maximum resolution supported is shown as -1x-1 in Xorg.log when
trying to run Xorg. This is occurring due to width and height variables
overflowing as they are assigned UINT_MAX instead of INT_MAX.
This patch fixes this by correcting the data types of width and height
variables and correctly assigning INT_MAX as maximum width and height.

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Signed-off-by: Saurabh Sengar <saurabh.singh@xilinx.com>
Reviewed-by: Sandip Kothari <sandipk@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/xlnx_crtc.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/xlnx/xlnx_crtc.c b/drivers/gpu/drm/xlnx/xlnx_crtc.c
index 8fc0282fdbd9..d5805c923675 100644
--- a/drivers/gpu/drm/xlnx/xlnx_crtc.c
+++ b/drivers/gpu/drm/xlnx/xlnx_crtc.c
@@ -49,7 +49,7 @@ struct xlnx_crtc_helper {
 	struct drm_device *drm;
 };
 
-#define XLNX_CRTC_MAX_HEIGHT_WIDTH	UINT_MAX
+#define XLNX_CRTC_MAX_HEIGHT_WIDTH	INT_MAX
 
 unsigned int xlnx_crtc_helper_get_align(struct xlnx_crtc_helper *helper)
 {
@@ -84,7 +84,7 @@ u64 xlnx_crtc_helper_get_dma_mask(struct xlnx_crtc_helper *helper)
 int xlnx_crtc_helper_get_max_width(struct xlnx_crtc_helper *helper)
 {
 	struct xlnx_crtc *crtc;
-	unsigned int width = XLNX_CRTC_MAX_HEIGHT_WIDTH, tmp;
+	int width = XLNX_CRTC_MAX_HEIGHT_WIDTH, tmp;
 
 	list_for_each_entry(crtc, &helper->xlnx_crtcs, list) {
 		if (crtc->get_max_width) {
@@ -99,7 +99,7 @@ int xlnx_crtc_helper_get_max_width(struct xlnx_crtc_helper *helper)
 int xlnx_crtc_helper_get_max_height(struct xlnx_crtc_helper *helper)
 {
 	struct xlnx_crtc *crtc;
-	unsigned int height = XLNX_CRTC_MAX_HEIGHT_WIDTH, tmp;
+	int height = XLNX_CRTC_MAX_HEIGHT_WIDTH, tmp;
 
 	list_for_each_entry(crtc, &helper->xlnx_crtcs, list) {
 		if (crtc->get_max_height) {
-- 
2.31.1

